### PR TITLE
feat(security): support transform CSS functions for sanitization.

### LINKED ISSUE #8514: More liberal style sanitization
Angular's current style sanitizer is very restrictive, effectively only allowing immediate primitive values and `rbg()`/`hsl()` pseudo calls. Angular should support additional common pseudo calls, such as the various animation properties `rotate`, `translate`, etc (common in dynamic bindings).

- [x] support common animation functions ([spec](https://www.w3.org/TR/css-transforms-1/#two-d-transform-functions))
- [ ] support `url()` by sanitizing its URL value ([spec](https://www.w3.org/TR/css3-values/#urls))

### PR DESCRIPTION
**Pull request by mprobst**
Fixes part of #8514.

### CODE REVIEW

**Comment by mprobst** [comment]
@rjamet PTAL?

File: modules/@angular/platform-browser/src/security/style_sanitizer.ts [hunk 1]
```diff
@@ -4,19 +4,24 @@ import {assertionsEnabled} from '../../src/facade/lang';
...
+ * transformation values.
  */
-const SAFE_STYLE_VALUE = /^([-,."'%_!# a-zA-Z0-9]+|(?:rgb|hsl)a?\([0-9.%, ]+\))$/;
+const VALUES = '[-,."\'%_!# a-zA-Z0-9]+';
+const TRANSFORMATION_FNS = '(?:matrix|translate|scale|rotate|skew|perspective)(?:X|Y|3d)?';
+const COLOR_FNS = '(?:rgb|hsl)a?';
+const FN_ARGS = '\\([-0-9.%, a-zA-Z]+\\)';
+
+const SAFE_STYLE_VALUE = new RegExp(
+    `^(` + VALUES + `|` + `(?:` + TRANSFORMATION_FNS + '|' + COLOR_FNS + ')' + FN_ARGS + `)$`, 'g');
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/security/style_sanitizer.ts]: any reason not to use string interpolation? `.... ${VALUES} ...`

**Comment by IgorMinar** [comment]
lgtm

**Comment by rjamet** [comment]
lgtm too

File: modules/@angular/platform-browser/src/security/style_sanitizer.ts (hunk 1 above)
Comment (mprobst) [inline modules/@angular/platform-browser/src/security/style_sanitizer.ts]: Done.

**Pull request merged**